sonar-helm-for-iac/make.sh

```diff
@@ -112,16 +112,20 @@ compile_binaries() {
   if [ -n "${GOOS:-}" ]; then
     # GOOS is already set, so we perform build for it
     echo "Building for target OS: ${GOOS}"
-    GO_FLAGS=(-ldflags="-s -w" -buildmode=c-shared)
+    GO_FLAGS=(-ldflags="-s -w" -buildmode=exe)
     case "${GOOS}" in
       "darwin")
         for GOARCH in amd64 arm64; do
-          CGO_ENABLED=1 go build "${GO_FLAGS[@]}" -o target/classes/sonar-helm-for-iac-"${GOOS}"-"${GOARCH}"
+          CGO_ENABLED=0 go build "${GO_FLAGS[@]}" -o target/classes/sonar-helm-for-iac-"${GOOS}"-"${GOARCH}"
         done
         ;;
-      "linux"|"windows")
+      "windows")
         GOARCH="amd64"
-        CGO_ENABLED=1 go build "${GO_FLAGS[@]}" -o target/classes/sonar-helm-for-iac-"$GOOS"-"$GOARCH"
+        CGO_ENABLED=0 go build "${GO_FLAGS[@]}" -o target/classes/sonar-helm-for-iac-"$GOOS"-"$GOARCH"
+        ;;
+      "linux")
+        GOARCH="amd64"
+        CGO_ENABLED=0 CC=musl-gcc go build "${GO_FLAGS[@]}" --ldflags '-linkmode external -extldflags "-s -w -static"' -o target/classes/sonar-helm-for-iac-"$GOOS"-"$GOARCH"
```

Comment:
Is static linking still needed if we build an executable instead of c-shared?

Reply:
Without static linking, we will have a runtime dependency on libc, which may once again cause differences on glibc- vs musl- based systems. Static linking with musl procudes a self-contained executable with little to no overhead in size